### PR TITLE
Ticket access

### PR DESCRIPTION
**Pull request by jean-neiverth**
We've decided to stop encrypting ticket access information, since we studied about signing and sending private messages and got to the conclusion it is too complex compared to the project's propose. We'll possibly come back to this later. Until then, the access data (like meeting links) will now be public and part of the event schema.

 So the changes are:
- Add  "access" attribute to the event schema 
- add QR Code in the ticket and a button to copy the access to clipboard

image

How to test:

1. Go to Events > New Event and create a new event with an access data
2. Go to the event details in Events > My Events > <your event>
3. Add a ticket to yourself
4. Reload the page and check your ticket

### CODE REVIEW

File: src/app/event/components/UserTicket.tsx [hunk 1]
```diff
@@ -13,6 +13,40 @@ export type TicketInfoType =
...
+}) => {
+  const [copied, setCopied] = useState(false);
+
+  const copyToClipboard = async (text: string) => {
+    await navigator.clipboard.writeText(text);
+    setCopied(true);
+    setTimeout(() => {
+      setCopied(false);
+    }, 2000);
+  };
```
Comment (ribeirojose) [inline src/app/event/components/UserTicket.tsx]: good candidate for a hook. I think we probably have one in fact to do this in @bleu/ui

File: src/app/events/new/components/FormField.tsx [hunk 2]
```diff
@@ -1,52 +1,53 @@
...
-	| "eventName"
-	| "startsAt"
-	| "startsAtTime"
-	| "endsAt"
-	| "endsAtTime"
-	| "type"
-	| "imageUrl"
-	| "briefDescription"
-	| "fullDescription";
+  | "eventName"
```
Comment (ribeirojose) [inline src/app/events/new/components/FormField.tsx:4]: formatting fix?

File: src/app/events/new/forms/Description.tsx [hunk 3]
```diff
@@ -65,6 +67,15 @@ export function Description() {
             </span>
           )}
         </div>
+        <FormField
+          className="w-1000px h-12 mt-1 pl-2 border-2 rounded-lg"
+          label="Access information"
+          type="text"
+          placeholder="Place information people will need to access the event (meeting links, QR code, etc.)"
```
Comment (ribeirojose) [inline src/app/events/new/forms/Description.tsx]: how would they place a QR code here? 🤔 pasting a QR code link?

**Review by ribeirojose — APPROVED** [review]

File: src/app/events/new/components/FormField.tsx (hunk 2 above)
Comment (jean-neiverth) [reply src/app/events/new/components/FormField.tsx:4]: yeah, some biome / prettier configs are probably different between mine and luiza's environment, but we are trying to fix this

File: src/app/events/new/forms/Description.tsx (hunk 3 above)
Comment (jean-neiverth) [reply src/app/events/new/forms/Description.tsx]: well observed

File: src/app/event/components/PageComponents.tsx [hunk 4]
```diff
@@ -79,7 +79,7 @@ export const OtherInfoSection = ({
               </TabsTrigger>
             ) : undefined}
           </TabsList>
-          <Modal eventId={eventId} />
+          {userIsEventOwner ? <Modal eventId={eventId} /> : undefined}
```
Comment (luizakp) [inline src/app/event/components/PageComponents.tsx:82]: prefer to use 

```
{userIsEventOwner && <Modal eventId={eventId} />}
```

**Pull request merged**